packages/cysync-core/src/pages/MainApp/Components/AssetAllocation.tsx

```diff
@@ -0,0 +1,171 @@
+import {
+  AllocationShare,
+  AssetIconNameBox,
+  NameBox,
+  NameVariants,
+  TableBody,
+  TableBox,
+  TableBoxDataRow,
+  TableBoxHeader,
+  TableBoxHeaderData,
+  TableBoxTitle,
+  Typography,
+} from '@cypherock/cysync-ui';
+import React, { useEffect, useState } from 'react';
+
+interface HeadersData {
+  name: string;
+  width?: string;
+  padding?: string;
+  $noFlex?: boolean;
+}
+
+const headersData: HeadersData[] = [
+  {
+    name: 'Asset',
+    padding: '16px 20px 16px 96px',
+    width: '300',
+    $noFlex: true,
+  },
+  {
+    name: 'Price',
+  },
+  {
+    name: 'Amount',
+  },
+  {
+    name: 'Value',
+  },
+  {
+    name: 'Allocation',
+    width: '300',
+    $noFlex: true,
+  },
+];
+
+interface AssetDataType {
+  name: NameVariants;
+  symbol: string;
+  price: string;
+  amount: string;
+  value: string;
+  allocation: number;
+}
+
+export const AssetAllocation = () => {
+  const [sortedBy, setSortedBy] = React.useState(headersData[0].name);
+  const [ascending, setAscending] = useState(true);
+  const [data, setData] = useState<AssetDataType[]>([
+    {
+      name: 'Bitcoin',
+      symbol: 'BTC',
+      price: '$ 16981.44',
+      amount: '0.0178 BTC',
+      value: '$2.9827',
+      allocation: 23.55,
+    },
+    {
+      name: 'Ethereum',
+      symbol: 'ETH',
+      price: '$ 674.44',
+      amount: '0.0179 ETH',
+      value: '$45.4527',
+      allocation: 74.47,
+    },
+  ]);
+
+  const handleClick = (columnName: string) => {
+    if (sortedBy === columnName) {
+      setAscending(!ascending);
+      return;
+    }
+    setSortedBy(columnName);
+    setAscending(true);
+  };
+
+  const comparator = (a: AssetDataType, b: AssetDataType) => {
+    if (sortedBy === 'Asset') {
+      return ascending
+        ? a.name.localeCompare(b.name)
+        : b.name.localeCompare(a.name);
+    }
+    if (sortedBy === 'Price') {
+      return ascending
+        ? parseFloat(a.price.replace('$ ', '')) -
+            parseFloat(b.price.replace('$ ', ''))
+        : parseFloat(b.price.replace('$ ', '')) -
+            parseFloat(a.price.replace('$ ', ''));
+    }
+    if (sortedBy === 'Amount') {
+      return ascending
+        ? parseFloat(a.amount.replace(' BTC', '')) -
+            parseFloat(b.amount.replace(' BTC', ''))
+        : parseFloat(b.amount.replace(' BTC', '')) -
+            parseFloat(a.amount.replace(' BTC', ''));
+    }
+    if (sortedBy === 'Value') {
+      return ascending
+        ? parseFloat(a.value.replace('$', '')) -
+            parseFloat(b.value.replace('$', ''))
+        : parseFloat(b.value.replace('$', '')) -
+            parseFloat(a.value.replace('$', ''));
+    }
+    if (sortedBy === 'Allocation') {
+      return ascending
+        ? a.allocation - b.allocation
+        : b.allocation - a.allocation;
+    }
```

Comment:
We can add the sorting algorithm function for each column to the `headersData` array, and use them here.